File: 0206. Reverse Linked List (Iteration)/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        output = None
        while head is not None:
            temp = head
            head = head.next
            temp.next = output
            output = temp
        return output
        

File: 0206. Reverse Linked List (Iteration)/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode output = null;
        while (head != null) {
            ListNode temp = head;
            head = head.next;
            temp.next = output;
            output = temp;
        }
        return output;
    }
}